Does not compile on Mac

I am trying to install podio, fcc-edm and fcc-physics on Mac. For each of these I have taken the master from Github and followed the README instructions.

Podio installs fine and tests run as expected.

fcc-edm fails to compile with two items like this.

[ 85%] Building CXX object utilities/CMakeFiles/utilities.dir/DummyGenerator.cc.o
/Users/alice/fccwork0.8/fcc-edm/utilities/DummyGenerator.cc:34:50: error: expected
      '(' for function-style cast or type construction
  writer.registerForWrite<fcc::ParticleCollection>("GenParticle");
                          ~~~~~~~~~~~~~~~~~~~~~~~^

--- a/examples/simplewrite.cc
+++ b/examples/simplewrite.cc
@@ -38,8 +38,8 @@ int main(){
   auto& evinfocoll = store.create<fcc::EventInfoCollection>("evtinfo");
   auto& pcoll = store.create<fcc::ParticleCollection>("mcparticles");
 
-  writer.registerForWrite<fcc::EventInfoCollection>("evtinfo");
-  writer.registerForWrite<fcc::ParticleCollection>("mcparticles");
+  writer.registerForWrite("evtinfo");
+  writer.registerForWrite("mcparticles");
 
   for(unsigned iev=0; iev<nevents; ++iev) {
     if(iev % 1000 == 0)

--- a/examples/write.cc
+++ b/examples/write.cc
@@ -54,7 +54,7 @@ int main(){
   unsigned nevents=10000;
 
   auto& evinfocoll = store.create<fcc::EventInfoCollection>("EventInfo");
-  writer.registerForWrite<fcc::EventInfoCollection>("EventInfo");
+  writer.registerForWrite("EventInfo");
 
   for(unsigned i=0; i<nevents; ++i) {
   // fill event information

--- a/init.sh
+++ b/init.sh
@@ -12,7 +12,7 @@ if [[ "$unamestr" == 'Linux' ]]; then
     if [[ -d /cvmfs/fcc.cern.ch/sw ]] ; then
         #should check domain to make sure we're at CERN
         #or is this software available somewhere in Lyon?
-        source /cvmfs/fcc.cern.ch/sw/0.8/init_fcc_stack.sh
+        source /cvmfs/fcc.cern.ch/sw/0.8.1/init_fcc_stack.sh
     else
         if [ ! -z "$PODIO" ]; then
             export LD_LIBRARY_PATH=$FCCEDM/lib:$PODIO/lib:$LD_LIBRARY_PATH

--- a/init.sh
+++ b/init.sh
@@ -23,6 +23,7 @@ elif [[ "$unamestr" == 'Darwin' ]]; then
     if [ ! -z "$PODIO" ]; then
         export DYLD_LIBRARY_PATH=$FCCEDM/lib:$DYLD_LIBRARY_PATH:$PODIO/lib
     fi
+    echo platform detected: $platform
 fi
 if [ -z "$PODIO" ]; then
     echo "[ERROR] no podio setup detected, make sure it is setup"
@@ -32,5 +33,3 @@ if [ -z "$FCCDAG" ]; then
     echo "[ERROR] no dag setup detected, make sure it is setup by setting the FCCDAG environment variable or adding it to your cmake prefix path"
     return 1
 fi
-
-echo platform detected: $platform

--- a/tests/graph_write_test.cc
+++ b/tests/graph_write_test.cc
@@ -19,8 +19,8 @@ int main() {
   auto& particleColl = store.create<fcc::MCParticleCollection>("mcparticles");
   auto& vtxColl = store.create<fcc::GenVertexCollection>("mcvertices");
 
-  writer.registerForWrite<fcc::MCParticleCollection>("mcparticles");
-  writer.registerForWrite<fcc::GenVertexCollection>("mcvertices");
+  writer.registerForWrite("mcparticles");
+  writer.registerForWrite("mcvertices");
 
   auto ptc = particleColl.create();
   auto vtxStartParent = vtxColl.create();

--- a/utilities/DummyGenerator.cc
+++ b/utilities/DummyGenerator.cc
@@ -31,8 +31,8 @@ DummyGenerator::DummyGenerator(int npart,
   m_particles(store.create<fcc::ParticleCollection>("GenParticle")),
   m_jets(store.create<fcc::JetCollection>("GenJet"))
   {
-  writer.registerForWrite<fcc::ParticleCollection>("GenParticle");
-  writer.registerForWrite<fcc::JetCollection>("GenJet");
+  writer.registerForWrite("GenParticle");
+  writer.registerForWrite("GenJet");
 }
 
 

